torch/csrc/jit/codegen/cuda/root_domain_map.cpp

```diff
@@ -93,6 +93,11 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
   if (SelectOp* sop = dynamic_cast<SelectOp*>(consumer_tv_->definition())) {
     selected_id = sop->getSelectAxis();
   }
+  IterDomain* idx_selected_id = nullptr;
+  if (IndexSelectOp* sop =
+          dynamic_cast<IndexSelectOp*>(consumer_tv_->definition())) {
+    idx_selected_id = sop->getSelectAxis();
+  }
```

Comment:
If we see this pattern often, we should consider adding some routines to handle both `SelectOp` and `IndexSelectOp` uniformly.

Reply:
For this case, SelectOp and IndexSelectOp have different behaviors, see https://github.com/csarofeen/pytorch/pull/2183/files#diff-147f701ca808989bf9ad700751c261c21fc30cea6c0f434001cdc4c2ce0be042R113-R119. For rest cases, like https://github.com/csarofeen/pytorch/pull/2183/files#diff-13bc8a76736a772a30a4cf860ec9527385f8c35d4c8481d9eeff654e6f5f8a5bR1225-R1252 , we should cosider adding some routines. Right now, I added a TODO. How about refining them once gather/scatter are added?

Reply:
Yeah, that's fine.